### PR TITLE
Add alerts and audit logger functionality

### PR DESCRIPTION
**Pull request by xpertforextradeinc**
This pull request introduces a new alerting and audit logging module for the backend, aimed at improving operational visibility and reliability. The module provides safe logging defaults and integrates with Slack for alert notifications, while gracefully handling cases where Slack is not configured.

Alerting and logging enhancements:

* Added `logEvent` function for structured, audit-friendly logging to stdout/stderr, ensuring logs are machine-parseable for later ingestion.
* Introduced `sendSlackAlert` function that sends alerts to Slack via webhook, with fallback behavior if the webhook URL is missing or sending fails.
* Ensured safe object stringification in logs and alerts using the `safeStringify` utility to prevent crashes from unstringifiable objects.
* Centralized alert and audit logic in `backend/alerts.js` for improved maintainability and clarity.

### CODE REVIEW

**Review by xpertforextradeinc — COMMENTED** [review]
Added backend/alerts.js with branded logAlert fallback

File: backend/alerts.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    });
+    fallback();
+    return { ok: false, error: err?.message || String(err) };
+  }
+}
+
+module.exports = {
+  logEvent,
+  sendSlackAlert,
+};
```
[the remark this answers is not in the document]
Comment (xpertforextradeinc) [reply backend/alerts.js]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/xpertforextradeinc/unionledger/pull/63#discussion_r2843476782)

**Comment by xpertforextradeinc** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/xpertforextradeinc/unionledger/pull/63#pullrequestreview-3843941985)

**Review by xpertforextradeinc — COMMENTED** [review]
Overall strong addition — the alert + audit logger module is clean, safe by default, and won’t break production if Slack isn’t configured. A few notes to tighten reliability and long‑term maintainability:

### ✅ What’s Good
- Safe fallback when `SLACK_WEBHOOK_URL` is missing — prevents runtime crashes.
- `logEvent()` produces machine‑parseable JSON, good for SIEM ingestion.
- Slack message formatting is readable and consistent.
- `safeStringify()` prevents circular‑reference crashes.

### ⚠️ Recommended Improvements
1. **Timeout Handling**
   - `axios.post` uses `{ timeout: 8000 }`, which is fine, but consider surfacing timeout explicitly in the returned object for better upstream debugging.

2. **Severity Normalization**
   - `severity` is free‑form. Consider validating against a fixed enum:  
     `['info', 'warning', 'error', 'critical']`.

3. **Context Size**
   - Large context objects could exceed Slack’s message limits.  
     Consider truncating or summarizing oversized payloads.

4. **Error Logging**
   - When Slack fails, `fallback()` logs a warning, but we may want to include the HTTP status code if available.

5. **Unit Tests**
   - This module is important for observability. Adding tests for:
     - missing webhook  
     - Slack success  
     - Slack failure  
     - stringify edge cases  
     would strengthen reliability.

### 👍 Final Verdict
Good structure and safe defaults. With the small improvements above, this module will be production‑ready and resilient.
